src/gtest/test_foundersreward.cpp

```diff
@@ -0,0 +1,64 @@
+#include <gtest/gtest.h>
+
+#include "chainparams.h"
+#include "utilstrencodings.h"
+#include <string>
+#include <set>
+
+// To run test:
+// ./zcash-gtest --gtest_filter="founders_reward_test.*"
+
+TEST(founders_reward_test, testnet) {
+    SelectParams(CBaseChainParams::TESTNET);
+
+    CChainParams params = Params();
+    
+    // For Testnet, the first address is derived from the old script. 
+    // FOUNDERS_REWARD_SCRIPT = a9146708e6670db0b950dac68031025cc5b63213a49187
+    //                address = 2N2e2FRfP9D1dRN1oRWkH7pbFM69eGNAuQ4
+    EXPECT_EQ(params.GetFoundersRewardScript(1), ParseHex("a9146708e6670db0b950dac68031025cc5b63213a49187"));
+    EXPECT_EQ(params.GetFoundersRewardAddress(1), "2N2e2FRfP9D1dRN1oRWkH7pbFM69eGNAuQ4");
+    size_t nInterval = params.GetConsensus().nSubsidyHalvingInterval;
+    
+    // If the block height parameter is out of bounds, there is an assert.
+    // Block height must be >0 and <nSubsidyHalvingInterval.
+    EXPECT_DEATH(params.GetFoundersRewardScript(0), "nHeight");
+    EXPECT_DEATH(params.GetFoundersRewardScript(nInterval), "nHeight");
+    EXPECT_DEATH(params.GetFoundersRewardAddress(0), "nHeight");
+    EXPECT_DEATH(params.GetFoundersRewardAddress(nInterval), "nHeight");
+    
+    // Check that three different addresses are used for testnet
+    std::set<std::string> addresses;
+    size_t n = 0;
+    for (int i=1; i<nInterval; i++) {
+        addresses.insert(params.GetFoundersRewardAddress(i));

```

Comment:
This runs tests over `TESTNET` params address vector. I'd like to see two more axes tested with these same tests:

* test all chain params address vectors.
* test some edge case address vectors, in case we need to alter the vectors in surprising ways:

  + what if the vector has only a single element? (we want that to degenerate to behaving like the pre-PR code.)
  + what if the vector has a large number of addresses, say `== nSubsidyHalvingInterval`? (That extreme seems ridiculous, but if the indexing logic performs correctly we can not worry about a realistic upper limit.)